cheri-architecture.tex

```diff
@@ -1,19 +1,17 @@
-%%%% University of Cambridge tech-report formatting; enable when producing
-%%%% tech-report versions of these documents; otherwise, disable.
-%\documentclass[12pt,twoside,openright,a4paper]{report}
-%\setlength{\oddsidemargin}{-0.4mm} % 25 mm left margin
-%\setlength{\evensidemargin}{\oddsidemargin}
-%\setlength{\textwidth}{160mm}      % 25 mm right margin
-%\setlength{\topmargin}{-5.4mm}     % 20 mm top margin
-%\setlength{\headheight}{5mm}
-%\setlength{\headsep}{5mm}
-%\setlength{\footskip}{10mm}
-%\setlength{\textheight}{237mm}     % 20 mm bottom margin
-%%%% .. or regular document
+\ifdefined\trformat
+\documentclass[12pt,twoside,openright,a4paper]{report}
+\setlength{\oddsidemargin}{-0.4mm} % 25 mm left margin
+\setlength{\evensidemargin}{\oddsidemargin}
+\setlength{\textwidth}{160mm}      % 25 mm right margin
+\setlength{\topmargin}{-5.4mm}     % 20 mm top margin
+\setlength{\headheight}{5mm}
+\setlength{\headsep}{5mm}
+\setlength{\footskip}{10mm}
+\setlength{\textheight}{237mm}     % 20 mm bottom margin
+\else
 \documentclass[12pt,letterpaper,twoside,openright,fleqn]{report}
 \usepackage{etex}
```

Comment:
I'm surprised this is a TR-vs-regular thing. cheri-architecture 97eaba24f17f5ebff1ea0c9da40b383e69c5aedd added it here (well, back then this block was in preamble.tex) as needed with tikz for some reason, but for the ISAv8 TR this was indeed commented out. I suspect this is because since ~2015 LaTeX has enabled the eTeX functionality itself so it only matters if building on an old version (perhaps Ubuntu 16.04 LTS (Xenial Xerus) didn't quite have the feature in its TeX Live distribution or Wes was using Debian 8 "Jessie" at the time).

Reply:
Do you mean that we should always be using the etex package?  I think if we want to move this, we should probably do it as a separate followup commit from this change.